api/envoy/extensions/geoip_providers/maxmind/v3/maxmind.proto

```diff
@@ -18,25 +18,31 @@ option (xds.annotations.v3.file_status).work_in_progress = true;
 
 // [#protodoc-title: MaxMind Geolocation Provider]
 // MaxMind geolocation provider :ref:`configuration overview <config_geoip_providers_maxmind>`.
-// At least one geolocation database path :ref:`city_db_path <envoy_v3_api_field_extensions.geoip_providers.maxmind.v3.MaxMindConfig.city_db_path>`,
-// :ref:`isp_db_path <envoy_v3_api_field_extensions.geoip_providers.maxmind.v3.MaxMindConfig.isp_db_path>` or
-// :ref:`anon_db_path <envoy_v3_api_field_extensions.geoip_providers.maxmind.v3.MaxMindConfig.anon_db_path>` must be configured.
+// At least one geolocation database path :ref:`db_path_city <envoy_v3_api_field_extensions.geoip_providers.maxmind.v3.MaxMindConfig.db_path_city>`,
+// :ref:`db_path_isp <envoy_v3_api_field_extensions.geoip_providers.maxmind.v3.MaxMindConfig.db_path_isp>` or
+// :ref:`db_path_asn <envoy_v3_api_field_extensions.geoip_providers.maxmind.v3.MaxMindConfig.db_path_asn>` or
+// :ref:`db_path_anon <envoy_v3_api_field_extensions.geoip_providers.maxmind.v3.MaxMindConfig.db_path_anon>` must be configured.
 // [#extension: envoy.geoip_providers.maxmind]
 
+// [#next-free-field: 6]
 message MaxMindConfig {
   // Full file path to the Maxmind city database, e.g. /etc/GeoLite2-City.mmdb.
   // Database file is expected to have .mmdb extension.
-  string city_db_path = 1 [(validate.rules).string = {pattern: "^$|^.*\\.mmdb$"}];
+  string db_path_city = 1 [(validate.rules).string = {pattern: "^$|^.*\\.mmdb$"}];
```

Comment:
These are breaking changes, and although the proto is declared as WIP, I wonder if it is worth making this change and breaking users.

Reply:
These changes come from this discussion:

Reply:
i might be missing something but why change the name here?  none of the other properties of the field have changed from what i can tell?

Reply:
To keep consistency with the previous change of all other db_path. The idea is to break for future releases on config parsing time.

Reply:
but why break future releases at config parsing time?  as far as i can tell, only one field was added, nothing changed about the existing fields other than their name.

Reply:
The isp_db_path was being used to parse the asn mmdb instead of the isp db. I believe this was wrong specified at the beginning of the development.  So to keep consistency of field names I'm renaming all the database fields as per @ravenblackx suggestion.

Reply:
ahh gotcha, now it makes sense, thanks!

Reply:
the other option for dealing with this is @adisuissa suggestion: keep these fields but deprecate them, and add new fields with the proper semantics.  since it's a WIP API, it's probably fine as is, but worth considering from the perspective of prod users of this extension.

Reply:
+1 to @abeyad, although if the question is whether to introduce new fields and deprecate the old ones, or rename - I slightly prefer renaming **if this filter isn't being used in prod**.
I'm not sure how widespread this is, but officially we can always say it's in WIP, so one shouldn't have used it in prod.

Reply:
@adisuissa @abeyad If renaming is OK, it means that the only one left will be _is_anon_ to be renamed to _anon_ and PR will be ready to merge.  Can I get your ack on this approach?

Reply:
this filter may be already used in prod at Uber, cc @hwyuan

Reply:
context from old convo: https://github.com/envoyproxy/envoy/pull/32490#discussion_r1603949514

Reply:
Sure thing. What is the process for renaming/deprecating fields?  At least one of the renaming  from `isp_db_path` to `db_path_asn` might be confusing, but it was wrongly named before. The others are not really necessary.

Reply:
Given we can't really remove fields from the API, since it's a WIP API, I'm in favor of just renaming the field assuming it won't cause much consternation to the team at Pinterest, let's wait for feedback from @hwyuan.

Reply:
since @hwyuan is not responding for last 2 weeks, should we move on with this PR?

Reply:
I'm preparing to push some commits today to introduce a runtime FF to keep the previous behavior for the isp_db.
```
// This feature can be removed once the geoip filter users confirm that they are using the correct
// behavior for the isp_db
FALSE_RUNTIME_GUARD(envoy_reloadable_features_geoip_maxmind_provider_isp_db_load_asn_db);
```